Add multi_assoc function

vichle opened the pull request:
This allows you to easily create arrays of associated objects.
 Example provided in README.

aexmachina commented on index.js:
```diff
@@ -40,6 +40,24 @@
       };
     };
 
+    factory.multi_assoc = function(numElements, name, key, attrs){
```
For consistency with the rest of the API this should be called `assocMany`.

aexmachina commented on index.js:
```diff
@@ -40,6 +40,24 @@
+      return function(callback) {
+        results = [];
+        for(var i = 0; i < numElements; ++i)
+          results[i] = null;
```
Braces here please :)

aexmachina commented on index.js:
```diff
@@ -40,6 +40,24 @@
+        results = [];
+        for(var i = 0; i < numElements; ++i)
+          results[i] = null;
+        asyncForEach(results, function(elem, cb, index){
```
Space before brace please.

aexmachina commented on index.js:
```diff
@@ -40,6 +40,24 @@
+          results[i] = null;
+        asyncForEach(results, function(elem, cb, index){
+          factory.create(name, attrs, function(err, doc) {
+            if(err) return cb(err);
```
Space before brace here.

vichle commented:
Thanks for the comments, I think I have addressed your concerns. I am quite new to JS and this is also my first contribution on GitHub so please bear with me :)

aexmachina commented on index.js:
```diff
@@ -40,6 +40,25 @@
       };
     };
 
+    factory.assocMany = function(numElements, name, key, attrs) {
```
This method should take args `function(name, key, num, attrsArray)` for consistency with the other methods. Sorry, I should have picked this up before.

aexmachina commented:
Otherwise, LGTM.

vichle commented:
I realized that I did not handle cases where the key argument was set. Also realised I re-invented the wheel and am now using the createMany function instead of creating many on my own :)

aexmachina commented on README.md:
```diff
@@ -42,10 +42,16 @@ factory.build('user', function(err, user) {
   // or factory.assoc('user') to return the user object itself.
   user_id: factory.assoc('user', 'id'),
+  // create array of associations using factory.assocMany
+  comments: factory.assocMany(3, 'comment') // Will contains 3 elements
```
This usage example needs to be updated too.

aexmachina commented on index.js:
```diff
@@ -40,6 +40,27 @@
+        }
+      }
+      return function(callback) {
+        factory.createMany(name, attrsArray, num,  function(err, docs) {
```
Two spaces should be one: `...num, function(err, docs) {`

aexmachina commented:
Nice one @vichle, just two minor fixes and I'll press the big green button :)

vichle commented:
Thanks for having me @aexmachina :)

aexmachina commented:
Nice one, I think this is the point where I need to redo the documentation. But this is fine for now. Thanks!

aexmachina commented:
Published v1.5.0
